### ISSUE TITLE
Highlighting Warning about vars in TDZ, which contains in computed handler

### ISSUE THREAD

**Issue by vasilyman** [vuejs/eslint-plugin-vue#2081]
**Please describe what the rule should do:**
Currently linter dosen't highlighting warning about the computed property, which handler includes variables and constants, defined below in the code. (if variables in TDZ)

**What category should the rule belong to?**

- [ ] Warns about a potential error (problem)

**Provide 2-3 code examples that this rule should warn about:**

```vue
// this have not warning highlight (b - is in TDZ);
const a = computed(() => b);

// this case have hidden error, also have not warning highlight
console.log(a.value);

const b = '5'
```

**Additional context**
sandbox:

https://stackblitz.com/edit/vue-1yhuke?file=src/App.vue

**Comment by FloEdelmann** [comment]
What does TDZ mean?

**Comment by vasilyman** [comment]
> What does TDZ mean?

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/let#temporal_dead_zone_tdz

**Comment by ota-meshi** [comment]
I think you can use eslint core's no-use-before-define rule.
https://eslint.org/docs/latest/rules/no-use-before-define

Even if you want to check more strictly, I don't want to implement that rule in this plugin as I don't think it's a vue specific issue.

**Comment by vasilyman** [comment]
Ok, thanks for idea
